File: 0163-Missing Ranges/0163-Missing Ranges.cpp
class Solution {
public:
    vector<string> findMissingRanges(vector<int>& nums, int lower, int upper) {
        vector<string> result;
        long long prev = lower - 1LL;
        for (int num : nums)
        {
            if (prev + 2 == num)
            {
                result.push_back(to_string(prev + 1));
            }
            else if (prev + 2 < num)
            {
                result.push_back(to_string(prev + 1) + "->" + to_string(num - 1));
            }
            prev = num;
        }
        
        if (prev + 1 == upper)
        {
            result.push_back(to_string(prev + 1));
        }
        else if (prev + 1 < upper)
        {
            result.push_back(to_string(prev + 1) + "->" + to_string(upper));
        }
        
        return result;
    }
};

File: 0101-0200/0163-Missing Ranges/0163-Missing Ranges.py
class Solution:
    def findMissingRanges(self, nums: List[int], lower: int, upper: int) -> List[str]:
        result = []
        prev = lower - 1
        for num in nums:
            if prev + 2 == num:
                result.append(str(prev + 1))
            elif prev + 2 < num:
                result.append(str(prev + 1) + '->' + str(num - 1))
            prev = num
        if prev + 1 == upper:
            result.append(str(prev + 1))
        elif prev + 1 < upper:
            result.append(str(prev + 1) + '->' + str(upper))
        
        return result
